src/store/actions/api.actions.js

```diff
@@ -130,156 +284,166 @@ export const fetchEndpointSchema = () => (dispatch) => {
     dispatch(getEndpointSchemaSuccess(endpointSchema)); // @TODO: REMOVE when endpoint will be ready
 };
 
-export const updateEndpoint = (pathname, api) => (dispatch) => {
-    dispatch(saveEndpointRequest());
-
-    return client.put(`apis${pathname}`, api)
-    .then((response) => {
-        dispatch(saveEndpointSuccess());
-        dispatch(openResponseModal({ // @FIXME: move to reducers
-            status: response.status,
-            message: 'Successfuly saved',
-            statusText: response.statusText,
-        }));
-    })
-    .catch((error) => {
-        if (error.response) {
-            dispatch(openResponseModal({
-                status: error.response.status,
-                statusText: error.response.statusText,
-                message: error.response.data,
-            }));
-        // The request was made and the server responded with a status code
-        // that falls out of the range of 2xx
-        // More info about error handling in Axios: https://github.com/mzabriskie/axios#handling-errors
-            // eslint-disable-next-line
-            console.error(error.response.data);
-        } else if (error.request) {
-        // The request was made but no response was received
-        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
-        // http.ClientRequest in node.js
-            // eslint-disable-next-line
-            console.log(error.request);
-        } else {
-        // Something happened in setting up the request that triggered an Error
-            // eslint-disable-next-line
-            console.log('Error', error.message);
-        }
-    });
-};
+export const preparePlugins = api => api.plugins.map(plugin => {
```

Comment:
This function should not be public/exported right? Only for internal usage

Reply:
it's exporting for future. to make it tested by **Jest**[https://facebook.github.io/jest/](https://facebook.github.io/jest/).

But I've got your point => it should be extracted to separate file and tested there, but not here.

Caraca! 🇧🇷

Reply:
Just as reminder for myself: **issue**[https://github.com/hellofresh/janus-dashboard/issues/40](https://github.com/hellofresh/janus-dashboard/issues/40)